### PR TITLE
test: switch table tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Table tests to React Testing Library.

### Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

File: packages/react/src/components/Table/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  const firstTableHeader = tableHeaders[0];
+  const firstTableCell = tableCells[0];
+
+  expect(table).toHaveClass('my-table');
+  expect(tableHead).toHaveClass('my-table-head');
+  expect(tableBody).toHaveClass('my-table-body');
+  expect(firstTableRow).toHaveClass('my-table-row');
+  expect(firstTableHeader).toHaveClass('my-table-header');
+  expect(firstTableCell).toHaveClass('my-table-cell');
+  expect(tableFooter).toHaveClass('my-table-footer');
```
Comment (scurker) [inline packages/react/src/components/Table/index.test.tsx:108]: Very minor, but we should check for both the default class and the inherited class:

```suggestion
  expect(table).toHaveClass('Table', 'my-table');
  expect(tableHead).toHaveClass('TableHead', 'my-table-head');
  expect(tableBody).toHaveClass('TableBody', 'my-table-body');
  expect(firstTableRow).toHaveClass('TableRow', 'my-table-row');
  expect(firstTableHeader).toHaveClass('TableHeader', 'my-table-header');
  expect(firstTableCell).toHaveClass('TableCell', 'my-table-cell');
  expect(tableFooter).toHaveClass('TableFooter', 'my-table-footer');
```

File: packages/react/src/components/Table/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        <TableRow>
+          <TableHeader sortDirection={'ascending'} onSort={() => null}>
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByRole('status').closest('.TableHeader--sort-ascending'));
```
Comment (scurker) [inline packages/react/src/components/Table/index.test.tsx:217]: I think we're missing checking for `ascending` on the status role here?

File: packages/react/src/components/Table/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByText('up and away')).toBeInTheDocument();
```
Comment (scurker) [inline packages/react/src/components/Table/index.test.tsx:237]: Small suggestion to better validate that we're setting the announcement correctly.

```suggestion
  expect(screen.getByRole('status')).toHaveTextContent('up and away')
```

File: packages/react/src/components/Table/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByText('down below')).toBeInTheDocument();
```
Comment (scurker) [inline packages/react/src/components/Table/index.test.tsx:258]: Same as above:

```suggestion
  expect(screen.getByRole('status')).toHaveTextContent('down below');
```

File: packages/react/src/components/Table/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+  const button = screen.getByRole('button');
+
+  await userEvent.click(button);
+  await waitFor(() => {
+    expect(button).toHaveFocus();
+  });
+});
+
+test('returns 0 axe violations', async () => {
```
Comment (scurker) [inline packages/react/src/components/Table/index.test.tsx:310]: This test looks good, we just need additional axe tests for sorted columns (none, ascending, descending).

**Review by scurker — CHANGES_REQUESTED** [review]
Great work! Just a few minor things to change.
